drivers/regulator/regulator_stm32_vrefbuf.c

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2025 CodeWrights GmbH
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_vrefbuf
+
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/drivers/reset.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/logging/log.h>
+#include <stm32_ll_system.h>
+
+LOG_MODULE_REGISTER(regulator_stm32_vrefbuf, CONFIG_REGULATOR_LOG_LEVEL);
+
+struct regulator_stm32_vrefbuf_voltage {
+	uint32_t vrs; /* LL_VREFBUF_VOLTAGE_SCALE<X> */
+	int32_t uv;   /* VREFBUF output in uV with this scale */
+};
+
+struct regulator_stm32_vrefbuf_data {
+	struct regulator_common_data common;
+};
+
+struct regulator_stm32_vrefbuf_config {
+	struct regulator_common_config common;
+	const struct stm32_pclken pclken[1];
+	const struct reset_dt_spec reset;
+	const struct regulator_stm32_vrefbuf_voltage *ref_voltages;
+	bool vrefp_output_enable;
+	uint8_t ref_voltage_count;
+};
+
+static int regulator_stm32_vrefbuf_enable(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	LL_VREFBUF_Enable();
+
+	return 0;
+}
+
+static int regulator_stm32_vrefbuf_disable(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	LL_VREFBUF_Disable();
+
+	return 0;
+}
+
+static int regulator_stm32_vrefbuf_list_voltage(const struct device *dev, unsigned int idx,
+						int32_t *volt_uv)
+{
+	const struct regulator_stm32_vrefbuf_config *config = dev->config;
+
+	if (idx >= config->ref_voltage_count) {
+		return -EINVAL;
+	}
+
+	*volt_uv = config->ref_voltages[idx].uv;
+
+	return 0;
+}
+
+static unsigned int regulator_stm32_vrefbuf_count_voltages(const struct device *dev)
+{
+	const struct regulator_stm32_vrefbuf_config *config = dev->config;
+
+	return config->ref_voltage_count;
+}
+
+static int regulator_stm32_vrefbuf_set_voltage(const struct device *dev, int32_t min_uv,
+					       int32_t max_uv)
+{
+	const struct regulator_stm32_vrefbuf_config *config = dev->config;
+	uint32_t best_vrs = 0;
+	int32_t best_voltage = INT32_MAX;
+
+	/* choose minimum possible voltage in range provided by the caller */
```

Comment:
At least:
```suggestion
	/* search closest voltage above minimum requested by the caller */
```

But I'd suggest the following to remove ambiguity
```suggestion
	/*
	 * Search closest voltage above minimum requested by the caller
	 * among all voltages we support. Note that we must check all
	 * configurations, as the order is not consistent across series
	 * (i.e., whether VREFBUF0 < VREFBUF1 or VREFBUF0 > VREFBUF1).
	 */
```